topdown/regex.go

```diff
@@ -46,7 +46,7 @@ func builtinRegexMatch(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Te
 	if err != nil {
 		return err
 	}
-	return iter(ast.BooleanTerm(re.Match([]byte(s2))))
+	return iter(ast.BooleanTerm(re.MatchString(string(s2))))
```

Comment:
Nit: This one we can keep as-is I think since we're converting anyways.

Reply:
Correct if I'm wrong. `s2` is a `ast.String` (which is actually a `string`). So casting `string(s2)` doesn't incur a run-time cost like `[]byte(s2)` does [^1]

https://github.com/open-policy-agent/opa/blob/e903f5d815d1dae1a37ddaaef7deb5628bf5fb50/ast/term.go#L763-L764

![image](https://github.com/open-policy-agent/opa/assets/20135478/e10e50ab-742b-4947-a3df-6d1a852a668a)

[^1]: https://stackoverflow.com/a/32253871/7902371